ft232/d2xx.py

```diff
@@ -111,19 +112,26 @@ class D2xx(io.RawIOBase):
                    PARITY_ODD  : 1,
                    PARITY_EVEN : 2}
 
-    def __init__(self, port=None, baudrate=9600, bytesize=8, parity='N',
+    def __init__(self, serial_number=None, description=None, baudrate=9600, bytesize=8, parity='N',
                  stopbits=1, timeout=10, xonxoff=0, rtscts=0,
                  writeTimeout=10):
         self._isopen = False
-        self.portstr = port
-        serial = c.create_string_buffer(port.encode())
+        self.snstr = serial_number
+        self.descstr = description
         self.handle = c.c_void_p()
-        status = d2xx.FT_OpenEx(serial, FT_OPEN_BY_SERIAL_NUMBER,
-                                c.byref(self.handle))
-        if status != FT_OK:
-            status = d2xx.FT_OpenEx(serial, FT_OPEN_BY_DESCRIPTION,
+        status = FT_INVALID_PARAMETER
+        if serial_number:
```

Comment:
How about just 

if serial_number:
     #open by serial number
else if description:
     #open by description
else: 
     #raise exception

Reply:
Hi,
As I said earlier, d2xx and libftdi has different ways to open devices. In addition to general d2xx ways (by_serial and by_description) libftdi can open devices by full match with serial and description.
I agree with you, to uniquify both libs, we should open devices only by serial or description.
The question is should we try to open by description if we failed to open by serial and both options had been specified?

Reply:
Oh, no, just open by serial or description. (serial first then description if both are specified). If the user specifies a serial and it fail because it doesn't match we do not want to then open by description.

Reply:
Done!